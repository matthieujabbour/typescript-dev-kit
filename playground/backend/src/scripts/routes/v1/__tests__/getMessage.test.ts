import getMessage from 'scripts/routes/v1/getMessage';
import { FastifyRequest, FastifyReply } from 'fastify';

jest.mock('fastify');

describe('routes/v1/getMessage', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('correctly handles request', () => {
    const send = jest.fn();
    getMessage.handler({} as FastifyRequest, { send } as unknown as FastifyReply);
    expect(getMessage.schema).toMatchSnapshot();
    expect(send).toHaveBeenCalledTimes(1);
    expect(send).toHaveBeenCalledWith({ message: 'WELCOME_MESSAGE' });
  });
});
