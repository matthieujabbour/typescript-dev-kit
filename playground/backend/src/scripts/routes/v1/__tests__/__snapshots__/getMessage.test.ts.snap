// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes/v1/getMessage correctly handles request 1`] = `
Object {
  "body": undefined,
  "response": Object {
    "200": Object {
      "properties": Object {
        "message": Object {
          "type": "string",
        },
      },
      "type": "object",
    },
  },
}
`;
